### PR TITLE
Add tensor minimum/maximum ops

### PR DESCRIPTION
**Pull request by yuecideng**
Add two operations `Maximum`, `Minimum` in tensor function. 

This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/5261)

### CODE REVIEW

File: cpp/tests/core/TensorFunction.cpp [hunk 1]
```diff
@@ -234,5 +234,71 @@ TEST_P(TensorFunctionPermuteDevices, Append) {
...
+    EXPECT_TRUE(output.AllClose(
+            core::Tensor::Init<float>({{4, 5}, {4, 7}}, device)));
+
+    // Special case: Different input shape.
+    input = core::Tensor::Init<float>({2, 4, 3}, device);
+    other = core::Tensor::Init<float>({{4, 5, 2}, {3, 5, 1}}, device);
+    output = core::Maximum(input, other);
+    EXPECT_TRUE(output.AllClose(
+            core::Tensor::Init<float>({{4, 5, 3}, {3, 5, 3}}, device)));
+}
```
Comment (reyanshsolis) [inline cpp/tests/core/TensorFunction.cpp:268]: at least for one case, we must also verify behavior over different dtypes such as int, uint, and bool. You may add this after the Special case.

Comment (yuecideng) [reply cpp/tests/core/TensorFunction.cpp:268]: Should we make `bool` dtype supported? I think `LogicalAnd` and `LogicalOr` are equivalent to `Minimum` and `Maximum` for `bool` respectively. @reyanshsolis

**Review by yxlao — CHANGES_REQUESTED** [review]
> Reviewed 5 of 7 files at r1, 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/5261)* status: 6 of 7 files reviewed, 4 unresolved discussions (waiting on @reyanshsolis, @yuecideng, and @yxlao)

___
*[`cpp/open3d/core/kernel/BinaryEW.h` line 77 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5rFETTA_K97yp14sZD:-N5rFETUC9c73_oBFJtg:bskj2tf) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/open3d/core/kernel/BinaryEW.h#L77)):*
> ```objc
> #endif
> 
> inline void Add(const Tensor& lhs, const Tensor& rhs, Tensor& dst) {
> ```

Clean up previous code, and remove all of these.
___
*[`cpp/open3d/core/kernel/BinaryEWCUDA.cu` line 61 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5rDqQLAOS4hkmbD-e7:-N5rDqQLAOS4hkmbD-e8:b-yo188j) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/open3d/core/kernel/BinaryEWCUDA.cu#L61)):*
> ```Cuda
>     const scalar_t* rhs_ptr = static_cast<const scalar_t*>(rhs);
>     if (*lhs_ptr > *rhs_ptr) {
>         *static_cast<scalar_t*>(dst) = *lhs_ptr;
> ```

Can we try:
https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INT.html#group__CUDA__MATH__INT



_Code quote:_
```Cuda
OPEN3D_HOST_DEVICE
```
___
*[`cpp/pybind/core/tensor_function.cpp` line 109 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5rDzmYEE2LMoRyzXGh:-N5rDzmYEE2LMoRyzXGi:bamoadg) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/pybind/core/tensor_function.cpp#L109)):*
> ```cpp
>           R"(Computes the element-wise maximum of input and other. The tensors 
> must have same data type and device.
>           
> ```

nit: remove space
___
*[`cpp/tests/core/TensorFunction.cpp` line 268 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5ekkVq0pXsgw0s5gfT-r2-268:-N5rF4Z35BveAjBvn9oT:b-sg3zk1) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/tests/core/TensorFunction.cpp#L268)):*
<details><summary><i>Previously, yuecideng (Yueci Deng) wrote…</i></summary><blockquote>

Should we make `bool` dtype supported? I think `LogicalAnd` and `LogicalOr` are equivalent to `Minimum` and `Maximum` for `bool` respectively. @reyanshsolis 
</blockquote></details>

Try directly use dispatch template with bool and see if result is correct

**Review by yuecideng — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/5261)* status: 1 of 8 files reviewed, 3 unresolved discussions (waiting on @reyanshsolis and @yxlao)

___
*[`cpp/open3d/core/kernel/BinaryEW.h` line 77 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5rFETTA_K97yp14sZD:-N5t4CAO8hR5EW8erYfC:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/open3d/core/kernel/BinaryEW.h#L77)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Clean up previous code, and remove all of these.
</blockquote></details>

Done.
___
*[`cpp/open3d/core/kernel/BinaryEWCUDA.cu` line 61 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5rDqQLAOS4hkmbD-e7:-N5t4G6a6M3jK9Wpjsep:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/open3d/core/kernel/BinaryEWCUDA.cu#L61)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Can we try:
https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INT.html#group__CUDA__MATH__INT
</blockquote></details>

Done.
___
*[`cpp/pybind/core/tensor_function.cpp` line 109 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5rDzmYEE2LMoRyzXGh:-N5t4KrO99Wv7p0vE5jL:b-896fix) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/pybind/core/tensor_function.cpp#L109)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

nit: remove space
</blockquote></details>

Done.
___
*[`cpp/tests/core/TensorFunction.cpp` line 268 at r2](https://reviewable.io/reviews/isl-org/open3d/5261#-N5ekkVq0pXsgw0s5gfT-r2-268:-N5t4OeHBx2amI2FbtOs:btw9dbt) ([raw file](https://github.com/isl-org/open3d/blob/ad9788fdf99157d467318571829f74ee77e23032/cpp/tests/core/TensorFunction.cpp#L268)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Try directly use dispatch template with bool and see if result is correct
</blockquote></details>

Use dispatch template with bool for `Max` and `Min` conditions. Also add more test cases for these two functions.

**Review by yxlao — APPROVED** [review]
LGTM after CI

> Reviewed 5 of 7 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d/5261#-:-N5tC8da1hW7wBf3VY2x:b-p43wb3)* status: all files reviewed, 1 unresolved discussion (waiting on @reyanshsolis)

**Pull request merged**